src/ngraph/frontend/onnx_import/op/hardmax.cpp

```diff
@@ -0,0 +1,68 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "hardmax.hpp"
+#include "exceptions.hpp"
+#include "ngraph/frontend/onnx_import/utils/eye.hpp"
+#include "ngraph/frontend/onnx_import/utils/reshape.hpp"
+#include "ngraph/op/argmax.hpp"
+#include "ngraph/op/embedding_lookup.hpp"
+#include "ngraph/op/util/reshape.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                NodeVector hardmax(const Node& node)
+                {
+                    const auto input = node.get_ng_inputs().at(0);
+                    const auto& input_shape = input->get_shape();
+                    auto axis = node.get_attribute_value<std::int64_t>("axis", 1);
+
+                    NGRAPH_CHECK(axis >= 0 && axis < input_shape.size(),
```

Comment:
Please use `ASSERT_VALID_ARGUMENT` here, to include the node information in the error message.

Please also check if it's possible to use the newer `NODE_VALIDATION_CHECK`  here to replace the older and ONNX-specific `ASSERT_VALID_ARGUMENT`.

Reply:
`NODE_VALIDATION_CHECK` expects an `ngraph::Node` so I couldn't use it but I've changed the assertion macro to the suggested `ASSERT_VALID_ARGUMENT`